roles/postgresql/defaults/main.yml

```
@@ -126,10 +126,13 @@ postgresql_additional_databases: []
 
 postgresql_databases: "{{ postgresql_generated_databases + postgresql_additional_databases }}"
 
+#
+# Hive metastore uses "public" schema hardcoded, no schema is created for Hive (issue #28)
+#
 postgresql_generated_schemas: >-
   [
     {%- for db in postgresql_databases -%}
-      {%- if db.owner is defined -%}
+      {%- if db.owner is defined and db.owner != postgresql_hive_user -%}
```

We should make this more generic

We should create a list such as `no_custom_schemas_users` and check with:
```suggestion
      {%- if db.owner is defined and not db.owner in no_custom_schemas_users -%}
```

---

I propose you add an attribute `schema_generated` to items in `postgresql_databases`. You have an example with `hba_entry_generated` here https://github.com/TOSIT-IO/tdp-collection-prerequisites/blob/466a9d19da2548b31fdbb38139bd5b705aac3444/roles/postgresql/defaults/main.yml#L150-L153